bigtable-client-core-parent/bigtable-hbase-integration-tests-common/src/test/java/com/google/cloud/bigtable/hbase/TestScan.java

```diff
@@ -302,6 +309,149 @@ public void testStartEndEquals() throws IOException {
     }
   }
 
+  @Test
+  public void testBasicReverseScan() throws IOException {
+    String prefix = "reverse_basic";
+    int rowsToWrite = 10;
+
+    // Initialize variables
+    Table table = getDefaultTable();
+
+    byte[][] rowKeys = new byte[rowsToWrite][];
+    rowKeys[0] = dataHelper.randomData(prefix);
+    for (int i = 1; i < rowsToWrite; i++) {
+      rowKeys[i] = rowFollowingSameLength(rowKeys[i - 1]);
+    }
+
+    byte[] qualifier = dataHelper.randomData("qual-");
+    byte[] value = dataHelper.randomData("value-");
+
+    ArrayList<Put> puts = new ArrayList<>(rowsToWrite);
+
+    // Insert some columns
+    for (int rowIndex = 0; rowIndex < rowsToWrite; rowIndex++) {
+      Put put = new Put(rowKeys[rowIndex]).addColumn(COLUMN_FAMILY, qualifier, value);
+      puts.add(put);
+    }
+    table.put(puts);
+
+    Scan scan = new Scan().setReversed(true).withStartRow(rowKeys[6]).withStopRow(rowKeys[2]);
```

Comment:
I'm a bit confused.. so the Scan start key needs to be < stop key? which is the opposite of the query?

Reply:
yep...HBase flips it around for Scans :(